test/LibraryManager.Test/Configuration/SettingsTest.cs

```diff
@@ -0,0 +1,202 @@
+        public void TryGetValue_ConfigSectionDoesNotExist_ReturnFalseWithEmptyString()
+        {
+            string fileText = @"<configuration></configuration>";
+            File.WriteAllText(TestFilePath, fileText);
+
+            bool success = new TestSettings(TestFilePath).TryGetValue("testKey", out string value);
+
+            Assert.IsFalse(success);
+            Assert.AreEqual(string.Empty, value);
+        }
+
+        [TestMethod]
+        public void TryGetValue_ValueDoesNotExist_ReturnFalseWithEmptyString()
+        {
+            string fileText = @"<configuration><config><add key=""otherValue"" value=""other"" /></config></configuration>";
+            File.WriteAllText(TestFilePath, fileText);
+
+            bool success = new TestSettings(TestFilePath).TryGetValue("testKey", out string value);
+
+            Assert.IsFalse(success);
+            Assert.AreEqual(string.Empty, value);
+        }
+
+        [TestMethod]
+        public void TryGetValue_ValueExists_ReturnTrueWithValue()
+        {
+            string fileText = @"<configuration><config><add key=""testKey"" value=""testValue"" /></config></configuration>";
+            File.WriteAllText(TestFilePath, fileText);
+
+            bool success = new TestSettings(TestFilePath).TryGetValue("testKey", out string value);
+
+            Assert.IsTrue(success);
+            Assert.AreEqual("testValue", value);
+        }
+
+        [TestMethod]
+        public void TryGetValue_ValueExistsInDifferentCase_ReturnTrueWithValue()
+        {
+            string fileText = @"<configuration><config><add key=""testKey"" value=""testValue"" /></config></configuration>";
+            File.WriteAllText(TestFilePath, fileText);
+
+            bool success = new TestSettings(TestFilePath).TryGetValue("TestKey", out string value);
+
+            Assert.IsTrue(success);
+            Assert.AreEqual("testValue", value);
+        }
+
+        [TestMethod]
+        public void SetValue_ConfigSectionDoesNotExist_CreateSectionWithValue()
+        {
+            new TestSettings(TestFilePath).SetValue("testKey", "testValue");
+
+            string contents = File.ReadAllText(TestFilePath);
+            string expected = @"<?xml version=""1.0"" encoding=""utf-8""?>
+<configuration>
+  <config>
+    <add key=""testKey"" value=""testValue"" />
+  </config>
+</configuration>";
+            Assert.AreEqual(expected, contents);
+        }
+
+        [TestMethod]
+        public void SetValue_ValueDoesNotExist_InsertValue()
+        {
+            string fileText = @"<configuration><config><add key=""otherKey"" value=""other"" /></config></configuration>";
+            File.WriteAllText(TestFilePath, fileText);
+
+            new TestSettings(TestFilePath).SetValue("testKey", "testValue");
+
+            string contents = File.ReadAllText(TestFilePath);
+            string expected = @"<?xml version=""1.0"" encoding=""utf-8""?>
+<configuration>
+  <config>
+    <add key=""otherKey"" value=""other"" />
```

Comment:
It's an implementation detail, even after the switch to JSON.  Yes, it could be implemented differently, but I wanted to verify that there isn't any other weird thing happening in this file (e.g. adding a new <config> node, etc).  Verifying the file contents explicitly, while coupling to the implementation specifics, gave a succint but in-depth validation.